api/HouseInput.cs

```diff
@@ -19,86 +22,77 @@ public class HouseInput : IDeviceInput
 {
 	public HouseInput(string house_info, string device_info)
 	{
+	
+		_houseInfo = house_info;
+		_deviceInfo = device_info;
+		_Http_Client = new HttpClient();
+
+		try
+		{
+			var house_addr = JObject.Parse(house_info);
+			var temp_url = house_addr.GetValue("house_url").ToObject<string>();
+			Debug.WriteLine(temp_url);
+			_Http_Client.Timeout = TimeSpan.FromSeconds(10);
+			_Http_Client.BaseAddress = new Uri(temp_url);
+		}
+		catch(JsonException exception)
+		{
+			throw new ArgumentException(exception.Message);
+		}
+
 	}
 
 	public bool read(Device dev)
 	{
-		var _ = readHelper(dev);
 
-		return true;
+		return read_driver(dev);
```

Comment:
After you call `read_driver()`, you need to call `Interfaces.UpdateDevice(dev, json, silence_io:true)` with the body of the GET response. The JSON body is in the `read_driver()` call, so either return it from `read_driver` or do the update inside that function.